src/components/TodoHeader.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { useState } from 'react';
+
+import { Todo } from '../types/Todo';
+import { ErrorMessages } from '../types/ErrorNessages';
+
+type Props = {
+  isTodoShow: boolean,
+  newTodoTitle: string,
+  handleNewTodoTitle: (
+    newTodoTitle: React.ChangeEvent<HTMLInputElement>
+  ) => void,
+  addNewTodo: (title: string) => Promise<Todo | null>,
+  checkAllTodos: () => void;
+};
+
+export const TodoHeader: React.FC<Props> = ({
+  isTodoShow,
+  newTodoTitle,
+  handleNewTodoTitle,
+  addNewTodo,
+  checkAllTodos,
+}) => {
+  const [todoAdd, setTodoAdd] = useState(false);
+
+  const submitForm = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setTodoAdd(true);
+
+    try {
+      const newTodo = await addNewTodo(newTodoTitle);
+
+      if (newTodo === null) {
+        throw new Error(ErrorMessages.AddError);
+      }
+    } catch (error) {
+      throw new Error(ErrorMessages.AddError);
```

Comment:
it doesn't make sense to throw error here, inside catch - you don't handle it anywhere else - it will cause a page crash

Reply:
it was in my previous task, so i just copy my old code

Reply:
in this case I agree with Taras. submitForm function isn't reused function where you can throw an error to an upper scope. So, if you'd like to handle an error you can just set this error to the state and display for a user